scripts/o3de/NewComponent/New_Component.py

```diff
@@ -0,0 +1,540 @@
+#
```

Comment:
I have started to look at the folder structure of `script` and it seems that your could fit within the `scripts/o3de/ui` with some adjustement so that `o3de.bat` can run it. Else I believe the `Tools` folder at the root could be a good fit as well with a new folder named `ClassCreationWizard` or similar

Reply:
Also I wonder what is the motivation to have that as a python script instead of a C++ widget directly in the editor

Reply:
I started down this path some time ago but haven't had time to develop it, sharing it here if it helps: https://github.com/o3de/o3de/compare/development...lsemp3d:o3de:extend_editor_cpp_frontend